src/main/java/io/nats/client/impl/NatsJetStream.java

```diff
@@ -155,12 +163,64 @@ public void deleteConsumer(String streamName, String consumer) throws IOExceptio
      * {@inheritDoc}
      */
     @Override
-    public ConsumerLister getConsumers(String streamName) throws IOException, JetStreamApiException {
+    public List<String> getConsumerNames(String streamName) throws IOException, JetStreamApiException {
+        return getConsumerNames(streamName, null);
+    }
+
+    @Override
+    public List<String> getConsumerNames(String streamName, String filter) throws IOException, JetStreamApiException {
+        String subj = String.format(JSAPI_CONSUMER_NAMES, streamName);
+
+        ConsumerNamesResponse cnr = new ConsumerNamesResponse();
+        while (cnr.hasMore()) {
+            Message resp = makeRequestResponseRequired(subj, cnr.nextJson(filter).getBytes(StandardCharsets.US_ASCII), requestTimeout);
+            cnr.update(extractJsonThrowOnError(resp));
```

Comment:
Might want to rename this to `add()`?  Looks like it's not idempotent when applied to the same set.